packages/did-ipfs-store/src/didStore.ts

```diff
@@ -54,9 +55,19 @@ export class DidStore implements IDidStore {
   }
 
   async get(cid: string): Promise<string> {
-    const { data: content } = await axios.get(`${this.url}/ipfs/${cid}`, {
-      headers: this.headers,
-    });
+    let content: any;
+    try {
+      ({ data: content } = await axios.get(`${this.url}/ipfs/${cid}`, {
+        headers: this.headers,
+      }));
+    } catch (e) {
+      const { response } = <AxiosError>e;
+      if (response?.status === 504) {
+        throw new ContentNotFound(cid, response.statusText);
```

Comment:
@JGiter why does `504`, which indicates a Gateway Timeout, correspond to `ContentNotFound`?
```suggestion
      if (response?.status === 504) {
        throw new ContentNotFound(cid, response.statusText);
```

Reply:
My goal is to handle case when content isn't persisted and therefore response is equivalent to 404. But I haven't found a way to check this. All methods to locate or return content respond with 504 if no data found within given timeout. Sinse we are migrating to cluster consisting from comparatively not large amount of nodes, the 504 is actually means that requested data not in cluster, and not that IPFS don't have enough time to find it

Reply:
@JGiter Okay, I'm opening to adding this potentially, but I just want to understand it a little bit more. Let me try to paraphrase the behaviour you've observed to make sure I understand it correctly:

When requesting a CID from the EnergyWeb IPFS Cluster:
- if the data is available, then it is being returned
- if the data is not available, then after some time, a `504` is returned

Is the above correct ⬆️ ?